lua/cfc_prop_restoration/core/sv/sv_adinterface.lua

```diff
@@ -0,0 +1,136 @@
+ADInterface = {}
+
+local areacopy_classblacklist = {
+    gmod_anchor = true
+}
+
+local function PlayerCanDupeCPPI( ply, ent )
+    if ent.DoNotDuplicate or areacopy_classblacklist[ent:GetClass()] or not IsValid( ent:GetPhysicsObject() ) or not duplicator.IsAllowed( ent:GetClass() ) then return false end
+    return ent:CPPIGetOwner() == ply
+end
+
+local phys_constraint_system_types = {
+    Weld = true,
+    Rope = true,
+    Elastic = true,
+    Slider = true,
+    Axis = true,
+    AdvBallsocket = true,
+    NoCollide = true,
+    Motor = true,
+    Pulley = true,
+    Ballsocket = true,
+    Winch = true,
+    Hydraulic = true,
+    WireMotor = true,
+    WireHydraulic = true
+}
+
+local function GroupConstraintOrder( ply, constraints )
```

Comment:
I imagine you didn't write this part?

You might consider extracting this out to an `include/module` or `lua/cfc_prop_resetoration/util/` or `lua/cfc_prop_restoration/lib/` with a comment saying where you got it from, who wrote it, etc.